### ISSUE TITLE
Cannot run a Process under a Thread when using tf.set_random_seed

### ISSUE THREAD

**Issue by hayj** [tensorflow/tensorflow#29681]
**System information**
- OS Platform and Distribution: Linux Ubuntu 16 and 18
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6.8

**No issue on:** tf 1.5.0, Python 3.6.8 and Ubuntu 16

**Issue**

When using `tf.set_random_seed` I cannot run a Process under a Thread

**Code**

```python
import tensorflow as tf
from threading import Thread
from multiprocessing import Process

def misc():
    print("misc")

def do():
    p = Process(target=misc)
    p.start()
    p.join()

def test():
    a = Thread(target=do)
    a.start()
    a.join()

if __name__ == '__main__':
    print("main start")
    test()
    print("first test done")
    tf.set_random_seed(0)
    test()
    print("second test done")
```

This script ouput:

```
main start
misc
first test done
```
 
And the process block.

**Expected output**

```
main start
misc
first test done
misc
second test done
```

**Comment by gadagashwini** [comment]
I could able to reproduce the above issue with Tensorflow 1.13.1 on my environment. Thanks!

**Comment by ymodak** [comment]
Apologies for the delay in response. This is fixed in TF 1.14.0 
However if you select gpu accelerator the code will produce required result in TF 1.13.1
